enumerable.rb

```diff
@@ -0,0 +1,200 @@
+        result = false unless yield(k, v)
+      end
+    end
+    result
+  end
+
+  def my_any?(pattern = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    current = self
+    result = false
+    if !block_given? && pattern.nil?
+      current.my_each do |item|
+        result = true if item
+      end
+    elsif pattern.is_a? Regexp
+      current.my_each do |item|
+        result = true if item =~ pattern
+      end
+    elsif pattern.is_a? Class
+      current.my_each do |item|
+        result = true if item.is_a? pattern
+      end
+    elsif pattern
+      current.my_each do |item|
+        result = true if item == pattern
+      end
+    elsif is_a? Array
+      current.my_each do |item|
+        result = true if yield(item)
+      end
+    elsif is_a? Hash
+      current.my_each do |k, v|
+        result = true if yield(k, v)
+      end
+    end
+    result
+  end
+
+  def my_none?(pattern = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    result = true
+    current = self
+    if !block_given? && pattern.nil?
+      current.my_each do |item|
+        result = false if item
+      end
+    elsif pattern.is_a? Class
+      current.my_each do |item|
+        result = false if item.is_a? pattern
+      end
+    elsif pattern.is_a? Regexp
+      current.my_each do |item|
+        result = false if item =~ pattern
+      end
+    elsif pattern.is_a? Array
+      current.my_each do |item|
+        result = false if yield(item)
+      end
+    elsif pattern
+      current.my_each do |item|
+        result = false if item == pattern
+      end
+    end
+    result
+  end
+
+  def my_count(parameter = nil)
+    count = 0
+    if block_given?
+      my_each { |i| count += 1 if yield(self[i]) }
+    elsif !parameter.nil?
+      my_each { |i| count += 1 if self[i] == parameter }
+    else
+      count = length
+    end
+    count
+  end
+
+  def my_map
+    return to_enum unless block_given?
+
+    i = 0
+    new_map = []
+    while i < size
+      new_map << yield(self[i])
+      i += 1
+    end
+    new_map
+  end
+
+  def my_inject(initial_value = nil, symbol = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    if !initial_value.nil? && !symbol.nil?
+      my_each { |num| initial_value = initial_value.method(symbol).call(num) }
+      initial_value
+    elsif !initial_value.nil? && initial_value.is_a?(Symbol) && symbol.nil?
+      memo, *remaining_elements = self
+      remaining_elements.my_each { |num| memo = memo.method(initial_value).call(num) }
+      memo
+    elsif !initial_value.nil? && initial_value.is_a?(Integer) && symbol.nil?
+      my_each { |num| initial_value = yield(initial_value, num) }
+      initial_value
+    elsif initial_value.nil? && symbol.nil?
+      initial_value, *remaining_elements = self
+      remaining_elements.my_each { |num| initial_value = yield(initial_value, num) }
+      initial_value
+    end
+  end
```

Comment:
- [ ] Enumerables my_inject combines all elements of enum by applying a binary operation, specified by a block
`[1, 1, 1, 1, 1, 1, 1, 1].my_inject{proc { |sum, n| sum + n }}` should return `#<Proc:0x000055a055166cf0@main.rb:127>`

- [ ] Enumerables `my_inject`, when a symbol is specified, combines each element of the collection by applying the symbol as a named method
`[1,2,3,4,5].my_inject(&:+)` should return `15`

Reply:
it is returning the specified value that is asked